src/parser/TripleComponent.h

```diff
@@ -175,6 +175,10 @@ class TripleComponent {
   }
   [[nodiscard]] Variable& getVariable() { return std::get<Variable>(_variant); }
 
+  bool isId() const { return std::holds_alternative<Id>(_variant); }
+  [[nodiscard]] const Id& getId() const { return std::get<Id>(_variant); }
+  [[nodiscard]] Id& getId() { return std::get<Id>(_variant); }
```

Comment:
We don't use nodiscard by default.
Only when it is really semantically subtle (e.g. RAII types like Lock-Guard).
That clang-tidy rule of "every const method should be nodiscard" doesn't really add much value.